### PR TITLE
initial commit - all complete

### PR DESCRIPTION
**Pull request by NTR-TheMekaVerse**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -11,46 +11,180 @@
...
+    // Convert string to coded string via changes below: 
+        // 'a' --> 4
+        // 'e' --> 3
+        // 'i' --> 1
+        // 'o' --> 0
+    // Replace function?
+// Output: secret coded version of the input string
+
+
+describe("matrixLingo", (string) => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:32]: Superb

File: code-challenges.test.js [hunk 2]
```diff
@@ -11,46 +11,180 @@
...
+    const secretCodeWord1 = "Lackadaisical"
+    const secretCodeWord2 = "Gobbledygook"
+    const secretCodeWord3 = "Eccentric"
+        expect(matrixLingo(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect(matrixLingo(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect(matrixLingo(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
+
+// ******* RED FAILURE *******
```
Comment (SunkissedQueen) [inline code-challenges.test.js:43]: wonderful

File: code-challenges.test.js [hunk 3]
```diff
@@ -11,46 +11,180 @@
...
+// ReferenceError: matrixLingo is not defined
 
 
 // b) Create the function that makes the test pass.
 
+matrixLingo = (string) => {
+    return string.replace(/a/gi, "4").replace(/e/gi, "3").replace(/i/gi, "1").replace(/o/gi, "0")
```
Comment (SunkissedQueen) [inline code-challenges.test.js:50]: Yes one liner return statement!!!

File: code-challenges.test.js [hunk 4]
```diff
@@ -11,46 +11,180 @@
...
+}
+
+// ******* GREEN FAILURE *******
+// Although my first though was to use a loop, the built-in string method (.replace()) can perform its actions on all of the instances of the letters (throughout the word and case insensitive) by using the '/gi' ending in the first param.
```
Comment (SunkissedQueen) [inline code-challenges.test.js:54]: Glad you tested the theory. good work

File: code-challenges.test.js [hunk 5]
```diff
@@ -11,46 +11,180 @@
...
+
+// ******* RED FAILURE *******
+// ReferenceError: arrayContains is not defined
+
 
 
 // b) Create the function that makes the test pass.
 
+arrayContains = (array, letter) => {
+    return includesLetter = array.filter((value) => value.includes(letter) || value.includes(letter.toUpperCase()))
```
Comment (SunkissedQueen) [inline code-challenges.test.js:103]: I love that this function is dynamic and considers different edge cases

File: code-challenges.test.js [hunk 6]
```diff
@@ -11,46 +11,180 @@
...
+    // arrayContains
+    //   ✓ takes in an array of words and a single letter and returns all the words that contain that particular letter (1 ms)
 
-// --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
+    // Test Suites: 1 passed, 1 total
+    // Tests:       1 passed, 1 total
+
+
+// ******* REFACTOR *******
+// I believe this may be the cleanest version of this function.
```
Comment (SunkissedQueen) [inline code-challenges.test.js:119]: very clean

File: code-challenges.test.js [hunk 7]
```diff
@@ -11,46 +11,180 @@
...
+        const hand3 = [5, 5, 5, 5, 4]
+        expect(fullHouseChecker(hand1)).toEqual(true)
+        expect(fullHouseChecker(hand2)).toEqual(false)
+        expect(fullHouseChecker(hand3)).toEqual(false)
+    })
+})
+
+
+// ******* RED FAILURE *******
+// ReferenceError: fullHouseChecker is not defined
```
Comment (SunkissedQueen) [inline code-challenges.test.js:151]: appreciate your pseudocode and that you include the test suite outcome

File: code-challenges.test.js [hunk 8]
```diff
@@ -11,46 +11,180 @@
...
+    // PASS  ./code-challenges.test.js
+    // fullHouseChecker
+    // ✓ takes in an array of 5 numbers and determines whether or not the array is a “full house” (1 ms)
+
+    // Test Suites: 1 passed, 1 total
+    // Tests:       1 passed, 1 total
+
+
+// ******* REFACTOR *******
+// There should be a way to reference each index in the array without instantiating variables but I couldn't figure out how to get it to work when establishing the conditionals.
```
Comment (SunkissedQueen) [inline code-challenges.test.js:185]: You definitely can complete this without the variables.

File: ruby-challenges.rb [hunk 9]
```diff
@@ -11,25 +13,96 @@
...
+
+def word_contains_letter(array, letter)
+    array.select { |array| array.include?(letter) }
+end
+
+p word_contains_letter(beverages_array, letter_o)
+# ["coffee", "soda water"]
+p word_contains_letter(beverages_array, letter_t)
+# ["tea", "water", "soda water"]
+
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:28]: You give such quality work. I appreciate how you show your actual outcomes.

File: ruby-challenges.rb [hunk 10]
```diff
@@ -11,25 +13,96 @@
...
+# 2) Create a method that takes in an array of numbers and returns the sum of the numbers. Use the test variables provided.
 
 nums_array1 = [42, 7, 27]
 # Expected output: 76
 
 nums_array2 = [25, 17, 47, 11]
 # Expected output: 100
 
+def sum_nums array
+    array.map(&:to_i).sum
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:41]: How would you explain this technique to someone who has never seen it before

File: ruby-challenges.rb [hunk 11]
```diff
@@ -11,25 +13,96 @@
...
+p my_bike.pedal_faster(10)
+# 10
+p my_bike.pedal_faster(18)
+# 28
+p my_bike.brake(5)
+# 23
+p my_bike.brake(25)
+# 0
+p my_bike.bike_info
+# "The [\"Ninja\", 2, 0] bike has 2 wheels and is going 0 mph"
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:107]: Let's look at the initialize method Line 73.  
Hard coding the wheels and the current speed are a good call since those are the default values. I would suggest not hard coding the model. 
When you are creating a new Hash, pass an argument.

File: interview-questions.md [hunk 12]
```diff
@@ -7,41 +7,41 @@ Then, research the question to expand on your answer. Even if you feel you have
```
Comment (SunkissedQueen) [inline interview-questions.md:7]: Hand clap. Just to further enhance this great approach to interview questions. Share examples that you help your cohort understand these topics, tell any analogies that you used, or how you related to a previous subject. You can even drop that "while volunteering as a Teacher Assistant, I...." Let these people know how awesome and considerate you are.

**Review by SunkissedQueen — APPROVED** [review]
Tameka, you are a stellar developer. Make sure you share those wonderful traits with an interviewer on how you share your knowledge with fellow developers. I left some tips on how to expand your awesomeness in coding. Please reach out if you have any questions.
